pandas/core/indexes/base.py

```diff
@@ -1221,6 +1221,18 @@ def copy(
         -----
         In most cases, there should be no functional difference from using
         ``deep``, but if ``deep`` is passed it will attempt to deepcopy.
+
+        Examples
+        --------
+        >>> idx = pd.Index(['a', 'b', 'c'])
+        >>> new_idx = idx.copy(name="idx2")
+        >>> ser = pd.Series([1, 2, 3], index=new_idx)
```

Comment:
Probably better to show that `idx is new_idx` (without changing the `name`) returns `False` instead